src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> last;
+    private Node<T> first;
+
     @Override
     public void add(T value) {
+        linkedLastItem(value);
     }
 
     @Override
     public void add(T value, int index) {
+        checkPositionIndex(index);
+        if (index == size) {
+            linkedLastItem(value);
+        } else {
+            linkBeforeItem(value, findNodeByIndex(index));
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i < list.size(); i++) {
+            linkedLastItem(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return findNodeByIndex(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> nodeByIndex = findNodeByIndex(index);
+        T oldValue = nodeByIndex.item;
+        nodeByIndex.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlinkItem(findNodeByIndex(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> node = first; node != null; node = node.next) {
+            if (node.item == object || (object != null && object.equals(node.item))) {
+                unlinkItem(node);
+                return true;
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private static class Node<E> {
+        private E item;
+        private Node<E> next;
+        private Node<E> prev;
+
+        Node(Node<E> prev, E item, Node<E> next) {
+            this.prev = prev;
+            this.item = item;
+            this.next = next;
+        }
+    }
+
+    private T unlinkItem(Node<T> foundNode) {
+        final T removeElement = foundNode.item;
+        Node<T> next = foundNode.next;
+        Node<T> prev = foundNode.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            foundNode.prev = null;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            foundNode.next = null;
+        }
+
+        foundNode.item = null;
+        size--;
+        return removeElement;
+    }
+
+    private void linkBeforeItem(T item, Node<T> foundNode) {
```

Comment:
Same as the `linkedLastItem()` method, it is used only in 1 place and it is `add(T value, int index)`, so I don't see any reason to move almost all of its functionality to a private method